src/rulesets/Base/array.jl

```diff
@@ -24,66 +24,165 @@ end
 ##### `hcat` (🐈)
 #####
 
-function rrule(::typeof(hcat), A::AbstractArray, Bs::AbstractArray...)
-    function hcat_pullback(Ȳ)
-        Xs = (A, Bs...)
-        ntuple(length(Bs) + 2) do full_i
-            full_i == 1 && return NoTangent()
-
-            i = full_i - 1
-            l = mapreduce(j->size(Xs[j], 2), Base.add_sum, 1:i-1; init=0)
-            u = l + size(Xs[i], 2)
-            dim = u > l + 1 ? (l+1:u) : u
-            # NOTE: The copy here is defensive, since `selectdim` returns a view which we can
-            # materialize with `copy`
-            copy(selectdim(Ȳ, 2, dim))
+# work around https://github.com/JuliaLang/julia/issues/40809
+_get(x::Tuple, i::Int, default) = i in 1:length(x) ? x[i] : default
```

Comment:
This could be replaced by https://github.com/JuliaLang/Compat.jl/pull/744

Reply:
Yes, let's do that! :)